File: src/main/java/io/github/i/nahornyi/dummy/sampler/javaapi/actions/DummyActionBuilder.java
package io.github.i.nahornyi.dummy.sampler.javaapi.actions;

import io.gatling.javaapi.core.ActionBuilder;
import io.gatling.javaapi.core.ChainBuilder;

public class DummyActionBuilder implements ActionBuilder {

    private final io.github.i.nahornyi.dummy.sampler.actions.DummyActionBuilder wrapped;

    public DummyActionBuilder(io.github.i.nahornyi.dummy.sampler.actions.DummyActionBuilder wrapped) {
        this.wrapped = wrapped;
    }

    @Override
    public io.gatling.core.action.builder.ActionBuilder asScala() {
        return wrapped;
    }

    @Override
    public ChainBuilder toChainBuilder() {
        return ActionBuilder.super.toChainBuilder();
    }
}

File: src/main/scala/io/github/i/nahornyi/dummy/sampler/actions/DummyActionBuilder.scala
package io.github.i.nahornyi.dummy.sampler.actions

import io.gatling.core.action.Action
import io.gatling.core.action.builder.ActionBuilder
import io.gatling.core.structure.ScenarioContext

final case class DummyActionBuilder(attributes: DummySamplerAttributes) extends ActionBuilder {
  override def build(ctx: ScenarioContext, next: Action): Action = DummyRequestAction(attributes, ctx, next)
}